SCALERS.md

```diff
@@ -0,0 +1,57 @@
+
+> 📝 You can find a full list of supported scalers in [our documentation](https://keda.sh/docs/latest/scalers/).
+
+Scalers are shipped with KEDA in two ways:
+
+- **Built-in Scalers** - Scalers are packaged and shipped as part of the core KEDA runtime
+- **External Scalers** - Scalers that are running outside of KEDA but are capable of being used to scale workloads with. Workloads can be scaled by using the "External" or "External Push" scalers which integrate with the external scalers by using gRPC to drive scale.
+
+### Choosing the right model for a scaler
+
+When introducing a new scaler, it is important to choose the right model for it - Should it be a built-in scaler or an external scaler?
+
+The general rules of thumb for built-in scalers are the following:
+
+- They must live in the [core KEDA repository](http://github.com/kedacore/keda)
+- It follows the [KEDA core release cycle](https://github.com/kedacore/keda/blob/main/ROADMAP.md#upcoming-release-cycles)
+- It has no additional infrastructure requirements
+- It is general-purpose and not specific to a single end-user
+- It is written in Go
+
+If all of the above are true, then it is a potential case for a new built-in scaler. We recommend opening a feature request to propose and discuss it before doing the actual implementation though. This is to avoid throwing things away if it does not meet the needs after all.
+
+### Requirements for a built-in scaler
+
+In order to contribute a scaler to KEDA's core, it must meet the following requirements:
+
+- A scaler MUST be written in Go
+- A scaler maintainer(s) MUST be assigned (individual(s)/company) and will be tagged for bug reports
+- A scaler MUST have e2e and unit tests to guarantee its quality
+- A scaler MUST be backwards compatible until the next KEDA major version
+- A scaler CAN be deprecated when:
+  - Maintainers believe it is best for the project, after a vote
+  - The scaler maintainer(s) are not responsive and support volume is too high
+  - The technology is proven to be no longer used/abandoned
+- Scaler MUST follow KEDA release cycles
+- A scaler MUST be supported, as per [our support policy](SUPPORT.md), by the scaler maintainer(s) and will be automatically assigned to PRs. This will be done by using GitHub's Codeowners.
+
+### Choosing a home for external scalers
+
+Given the nature of external scalers, we recommend that they are maintained by the community and have a home outside of the KEDA organization. The reason for this is that it sets clear expectations for the community in terms of ownership and support.
+
+There have been exceptions where they have been added to the KEDA organization, but that is because they are governed and co-maintained by the community. Given that they are part of the KEDA organization, the maintainers are responsible for the quality of the scaler, timelines, etc.
```

Comment:
```
Given that they are part of the KEDA organization, the maintainers are responsible for the quality of the scaler, timelines, etc.
```

How should I understand this? KEDA core maintainers are responsible for these scalers? For example this scaler: https://github.com/kedacore/external-scaler-azure-cosmos-db

Reply:
In theory yes, I wrote this going forward but maybe it's time to evaluate if we should move that out into a Microsoft-owned organization.

I think the HTTP add-on is a better example.

However "quality" might be the wrong term here but more around release timelines, quality bars, etc and that we have the mandate to "enforce" things (ie end "responsibility/ownership")

Reply:
I am okay with keeping those external scalers in KEDA repo, but would rephrase the sentence I have highlighted. Core maintainers shouldn't be responsible for that. Maintainers of that repo should be responsible, if there are no maintainers, then it should be community best effort.

Reply:
Updated, let me know what you think